tensorflow_addons/layers/spectral_normalization.py

```diff
@@ -121,7 +121,10 @@ def normalize_weights(self):
 
             sigma = tf.matmul(tf.matmul(v, w), u, transpose_b=True)
 
-            self.w.assign(self.w / sigma)
+            if self.layer.compute_dtype == self.layer.dtype:
+                self.w.assign(self.w / sigma)
```

Comment:
So the point here with mixed precision is that the division between self.w (AutocastCastVariable) and sigma return e.g. a `tf.foat16` with the autocast overloaded div at
https://github.com/tensorflow/tensorflow/blob/bca08b32c1810dccc76a8f2bc3751b50a4269561/tensorflow/python/keras/mixed_precision/autocast_variable.py#L420-L421 
But then we cannot assign a `tf.float16` to `self.w` that is an `AutocastCastVariable` `dtype=float32 dtype_to_cast_to=float16`

It seems to me a quite general issue with wrapping layers and mixed precision when you need to manipulate weights.